interfaces/builtin/docker.go

```diff
@@ -0,0 +1,185 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var dockerPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the Docker daemon. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# Allow sockets
+/{,var/}run/docker.sock rw,
+/{,var/}run/docker/     rw,
+/{,var/}run/docker/**   mrwklix,
+/{,var/}run/runc/       rw,
+/{,var/}run/runc/**     mrwklix,
+
+# Wide read access to /proc, but somewhat limited writes for now
+@{PROC}/ r,
+@{PROC}/** r,
+@{PROC}/[0-9]*/attr/exec w,
+@{PROC}/sys/net/** w,
+@{PROC}/[0-9]*/cmdline r,
+@{PROC}/[0-9]*/oom_score_adj w,
+
+# Wide read access to /sys
+/sys/** r,
+# Limit cgroup writes a bit
+/sys/fs/cgroup/*/docker/   rw,
+/sys/fs/cgroup/*/docker/** rw,
+/sys/fs/cgroup/*/system.slice/   rw,
+/sys/fs/cgroup/*/system.slice/** rw,
+
+# Allow tracing other processes (especially ourselves)
+ptrace (read, trace),
+
+# Docker needs a lot of caps, but limits them in the app container
+capability,
+
+# Allow talking to systemd
+dbus (send)
+    bus=system
+    peer=(name=org.freedesktop.systemd*,label=unconfined),
+# Allow receiving from unconfined
+dbus (receive)
+    bus=system
+    peer=(label=unconfined),
+
+# Docker does all kinds of mounts all over the filesystem
+/dev/mapper/control rw,
+/dev/mapper/docker* rw,
+/dev/loop* r,
+/dev/loop[0-9]* w,
+mount,
+umount,
+pivot_root,
+/.pivot_root*/ rw,
+
+# for console access
+/dev/ptmx rw,
+# file descriptors (/proc/NNN/fd/X)
+/[0-9]*   rw,
```

Comment:
Yeah, I can reproduce and paste them, although maybe this will help (from inside a container):
```console
root@977b9f2d78fd:/# ls -l /proc/1/fd
total 0
lrwx------ 1 root root 64 Aug  2 20:23 0 -> /30
lrwx------ 1 root root 64 Aug  2 20:23 1 -> /30
lrwx------ 1 root root 64 Aug  2 20:23 2 -> /30
lrwx------ 1 root root 64 Aug  2 20:23 255 -> /30
```
(inside Docker containers, file descriptors end up showing as numbered files in `/` for some reason :smile:)